ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java

```diff
@@ -534,19 +538,19 @@ public void run() {
           OptimizerKeepingTask keepingTask = suspendingQueue.take();
           String token = keepingTask.getToken();
           boolean isExpired = !keepingTask.tryKeeping();
-          Optional.ofNullable(keepingTask.getQueue())
-              .ifPresent(
-                  queue ->
-                      queue
-                          .collectTasks(buildSuspendingPredication(token, isExpired))
-                          .forEach(task -> retryTask(task, queue)));
           if (isExpired) {
             LOG.info("Optimizer {} has been expired, unregister it", keepingTask.getOptimizer());
             unregisterOptimizer(token);
           } else {
             LOG.debug("Optimizer {} is being touched, keep it", keepingTask.getOptimizer());
             keepInTouch(keepingTask.getOptimizer());
           }
+          Optional.ofNullable(keepingTask.getQueue())
```

Comment:
I don't quite get the difference.

I would appreciate if you could explain more in this related issue/
Thanks.